File: fortune-reactive-postgresql/src/main/java/me/escoffier/fortune/reactive/postgresql/FortuneResource.java
package me.escoffier.fortune.reactive.postgresql;

import io.smallrye.mutiny.Uni;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import java.util.List;

@Path("/fortunes")
public class FortuneResource {


    private final FortuneRepository repository;

    public FortuneResource(FortuneRepository repository) {
        this.repository = repository;
    }

    @GET
    public Uni<List<Fortune>> getAllFortunes() {
        return this.repository.listAll();
    }

    @GET
    @Path("/random")
    public Uni<Fortune> getRandomFortune() {
        return this.repository.getRandomFortune();
    }

}

File: fortune-kotlin-postgresql/src/main/kotlin/me/escoffier/fortune/kotlin/postgresql/FortuneResource.kt
package me.escoffier.fortune.kotlin.postgresql

import io.smallrye.mutiny.coroutines.awaitSuspending
import me.escoffier.fortune.kotlin.postgresql.Fortune
import me.escoffier.fortune.kotlin.postgresql.FortuneRepository
import jakarta.enterprise.inject.Default
import jakarta.inject.Inject
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path

@Path("/fortunes")
class FortuneResource {

    @Inject
    @field: Default
    lateinit var repository: FortuneRepository;


    @GET
    suspend fun getAllFortunes(): List<Fortune> {
        return repository.listAll().awaitSuspending();
    }

    @GET
    @Path("/random")
    suspend fun getRandomFortune(): Fortune {
        return repository.randomFortune.awaitSuspending();
    }
}